bundles/http/Citrix/Covid-19-Self-Certify/script.js

```diff
@@ -0,0 +1,219 @@
+        return {responseBody:body,errorMessage:'Unable to process the request',errorStatus:true}
+    }
+}
+
+async function syncApp({ client, dataStore, integrationParameters }) {
+    await Promise.all([
+        getItems(client, dataStore, integrationParameters, false),
+        getRegionData(client, dataStore, integrationParameters)
+    ])
+}
+
+async function getItems(client, dataStore, integrationParameters, dataUpdateAfterAction = false) {
+    const limit = 100
+    let offset = 0
+    let total = 0
+    const userEmails = new Set()
+    do {
+        const itemsResponse = await client.fetch(`item/app/${integrationParameters.covidAppId}/filter/${integrationParameters.viewId}`, {
+            method: "POST",
+            body: JSON.stringify({
+                limit: limit,
+                offset: offset,
+                sort_by: "app_item_id",
+                sort_desc: true
+            })
+        })
+        const {responseBody,errorMessage,errorStatus} = await validateResponse(itemsResponse)
+        if (errorStatus) {
+            console.log(JSON.stringify(responseBody),errorMessage)
+            throw new Error(errorMessage)
+        }
+        total = responseBody.total
+        const items = responseBody.items ?? []
+        items.filter(item => {
+            const email = getTextValue(item.fields, 'email')
+            if (!userEmails.has(email)) {
+                userEmails.add(email)
+                return true
+            }
+            return false
+        }).map(item => {
+            const fields = item.fields ?? []
+            dataStore.save("items", {
+                "employeeName": getTextValue(item.fields, 'text', 'Name') ?? null,
+                "email": getTextValue(item.fields, 'email') ?? null,
+                "confirmation": getCategoryValue(item.fields) ?? null,
+                "createdOn": getTimestamp(item.fields) ?? null,
+                "app_item_id": item.app_item_id,
+                "region": getTextValue(item.fields, 'text', 'Region') ?? null,
+                "country": getTextValue(item.fields, 'text', 'Country') ?? null,
+            })
+        })
+        offset += limit
+    } while (dataUpdateAfterAction != true && offset <= total)
+}
+
+async function covidSelfCertify({ client, dataStore, integrationParameters, actionParameters }) {
+    if (actionParameters.appId !== null) {
+        dataStore.deleteById('items', actionParameters.email)
+    }
+    const response = await client.fetch(`item/app/${integrationParameters.covidAppId}`, {
+        method: "POST",
+        body: JSON.stringify({
+            "fields": {
+                "email": [{ "type": "work", "value": actionParameters.email }],
+                "title": [{ "value": actionParameters.email.substring(0, actionParameters.email.indexOf('@')).split('.').reduce((initialvalue, currentvalue) => { return initialvalue + (currentvalue[0].toUpperCase() + currentvalue.substring(1, currentvalue.length) + ' ') }, '').trimEnd() }],
```

Comment:
line too long: refactor title extraction to dedicated function

Reply:
Implemented a new function `getTitleFromEmail`.